src/librustc/ty/flags.rs

```diff
@@ -85,6 +85,15 @@ impl FlagComputation {
                 }
             }
 
+            &ty::TyGenerator(_, ref substs, ref interior) => {
+                // FIXME: Find out why TyClosure has HAS_TY_CLOSURE
+                // and see if the same reason applies here
```

Comment:
It's because trait selection etc. on a closure type depends on the current typeck tables status. Which is also the situation with generators.